workers.go

```diff
@@ -186,19 +227,31 @@ func (api *API) UploadWorker(requestParams *WorkerRequestParams, data string) (W
 	return r, nil
 }
 
-// UploadWorkerWithName push raw script content for your worker
-// This is an enterprise only feature https://developers.cloudflare.com/workers/api/config-api-for-enterprise/
-//
-// API reference: https://api.cloudflare.com/#worker-script-upload-worker
-func (api *API) uploadWorkerWithName(scriptName string, data string) (WorkerScriptResponse, error) {
-	if api.OrganizationID == "" {
-		return WorkerScriptResponse{}, errors.New("organization ID required for enterprise only request")
+func (api *API) doMultipartUploadWorker(uri, data string, metadata WorkerMetadataParams) (WorkerScriptResponse, error) {
+	var r WorkerScriptResponse
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	metadataPart, err := writer.CreateFormFile("metadata", "metadata.json")
+	if err != nil {
+		return r, errors.Wrap(err, errMakeRequestError)
+	}
+	b, err := json.Marshal(metadata)
+	if err != nil {
+		return r, errors.Wrap(err, errMakeRequestError)
+	}
+	io.Copy(metadataPart, bytes.NewReader(b))
+	scriptPart, err := writer.CreateFormFile("script", "script.js")
+	if err != nil {
+		return r, errors.Wrap(err, errMakeRequestError)
+	}
+	io.Copy(scriptPart, bytes.NewReader([]byte(data)))
+	err = writer.Close()
+	if err != nil {
+		return r, errors.Wrap(err, errMakeRequestError)
 	}
-	uri := "/accounts/" + api.OrganizationID + "/workers/scripts/" + scriptName
 	headers := make(http.Header)
-	headers.Set("Content-Type", "application/javascript")
-	res, err := api.makeRequestWithHeaders("PUT", uri, []byte(data), headers)
-	var r WorkerScriptResponse
+	headers.Set("Content-Type", writer.FormDataContentType())
+	res, err := api.makeRequestWithHeaders("PUT", uri, body.Bytes(), headers)
```

Comment:
should we consider switching this to [`http.MethodPut`](https://golang.org/pkg/net/http/#pkg-constants) too to make it more consistent with line 219?